x/staking/keeper/validator.go

```diff
@@ -138,12 +138,43 @@ func (k Keeper) UpdateValidatorCommission(ctx sdk.Context,
 		return commission, err
 	}
 
+	if newRate.LT(k.MinCommissionRate(ctx)) {
+		return commission, fmt.Errorf("cannot set validator commission to less than minimum rate of %s", k.MinCommissionRate(ctx))
+	}
+
 	commission.Rate = newRate
 	commission.UpdateTime = blockTime
 
 	return commission, nil
 }
 
+// MustUpdateValidatorCommission updates a validator's commission rate,
+// ignoring limitations (update-time, max-rate and max-change-rate)
+// This shouldn't be used in normal cases. Only for chain upgrade handlers.
+func (k Keeper) MustUpdateValidatorCommission(ctx sdk.Context,
+	validator types.Validator, newRate sdk.Dec) (types.Commission, error) {
+	commission := validator.Commission
+	blockTime := ctx.BlockHeader().Time
+
+	// validate newRate
+	switch {
+	case newRate.IsNegative():
+		return commission, types.ErrCommissionChangeRateNegative
+	case newRate.GT(sdk.OneDec()):
+		return commission, fmt.Errorf("cannot set validator commission to larger than 100%%")
```

Comment:
```suggestion
		return commission, fmt.Errorf("cannot set validator commission to larger than 100%")
```

Reply:
I had to escape the `%` with `%` because the `fmt.Errorf()` understands `%` as a special character for string interpolation. 
https://go.dev/play/p/zys4wDhEe3p
If it looks weird, I can replace `fmt.Errorf()` with `errors.New()` but I didn't because of the consistency with the next line.